pkg/fixtures/writer_ddb_kvstore.go

```diff
@@ -3,49 +3,66 @@ package fixtures
 import (
 	"context"
 	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/ddb"
 	"github.com/applike/gosoline/pkg/kvstore"
 	"github.com/applike/gosoline/pkg/mdl"
 	"github.com/applike/gosoline/pkg/mon"
 )
 
+type ddbKvstoreFactory func() kvstore.KvStore
+
 type KvStoreFixture struct {
 	Key   interface{}
 	Value interface{}
 }
 
 type dynamoDbKvStoreFixtureWriter struct {
-	logger mon.Logger
-	store  kvstore.KvStore
+	logger  mon.Logger
+	factory ddbKvstoreFactory
+	purger  *dynamodbPurger
 }
 
 func DynamoDbKvStoreFixtureWriterFactory(modelId *mdl.ModelId) FixtureWriterFactory {
 	return func(config cfg.Config, logger mon.Logger) FixtureWriter {
-		store := kvstore.NewDdbKvStore(config, logger, &kvstore.Settings{
-			AppId: cfg.AppId{
-				Project:     modelId.Project,
-				Environment: modelId.Environment,
-				Family:      modelId.Family,
-				Application: modelId.Application,
-			},
-			Name: modelId.Name,
+		factory := func() kvstore.KvStore {
```

Comment:
i do not get the reason why we need a factory here

Reply:
solved in private conversation